client/driver/rkt.go

```diff
@@ -435,66 +472,181 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
+		}
+
+		// Add memory isolator
+		prepareArgs = append(prepareArgs, fmt.Sprintf("--memory=%vM", int64(task.Resources.MemoryMB)))
 
-	// Add memory isolator
-	prepareArgs = append(prepareArgs, fmt.Sprintf("--memory=%vM", int64(task.Resources.MemoryMB)))
+		// Add CPU isolator
+		prepareArgs = append(prepareArgs, fmt.Sprintf("--cpu=%vm", int64(task.Resources.CPU)))
+
+	} else {
+		allocACName, err := appctypes.NewACName(allocVolName)
+		if err != nil {
+			return nil, fmt.Errorf("[ERR] driver.rkt: failed to set ACName: %s", err)
+		}
 
-	// Add CPU isolator
-	prepareArgs = append(prepareArgs, fmt.Sprintf("--cpu=%vm", int64(task.Resources.CPU)))
+		localACName, err := appctypes.NewACName(localVolName)
+		if err != nil {
+			return nil, fmt.Errorf("[ERR] driver.rkt: failed to set ACName: %s", err)
+		}
+
+		secretsACName, err := appctypes.NewACName(secretsVolName)
+		if err != nil {
+			return nil, fmt.Errorf("[ERR] driver.rkt: failed to set ACName: %s", err)
+		}
+
+		podManifest.Volumes = []appctypes.Volume{
+			{
+				Name:   *allocACName,
+				Kind:   "host",
+				Source: ctx.TaskDir.SharedAllocDir,
+			},
+			{
+				Name:   *localACName,
+				Kind:   "host",
+				Source: ctx.TaskDir.SharedAllocDir,
+			},
+			{
+				Name:   *secretsACName,
+				Kind:   "host",
+				Source: ctx.TaskDir.SharedAllocDir,
+			},
+		}
+
+		mounts := []appcschema.Mount{
+			{
+				Volume: *allocACName,
+				Path:   allocdir.SharedAllocContainerPath,
+			},
+			{
+				Volume: *localACName,
+				Path:   allocdir.TaskLocalContainerPath,
+			},
+			{
+				Volume: *secretsACName,
+				Path:   allocdir.TaskSecretsContainerPath,
+			},
+		}
+
+		var environment appctypes.Environment
+		for name, value := range ctx.TaskEnv.Map() {
+			environment = append(environment, appctypes.EnvironmentVariable{
+				Name:  name,
+				Value: value,
+			})
+		}
+
+		fetchArgs := make([]string, 0, 50)
+		fetchArgs = append(fetchArgs, globalArgs...)
+		fetchArgs = append(fetchArgs, "fetch")
+
+		for i, app := range podManifest.Apps {
+			ID, err := d.fetchImage(app.Image, fetchArgs)
+			if err != nil {
+				return nil, err
+			}
+
+			currApp := &podManifest.Apps[i]
+
+			d.logger.Printf("[DEBUG] driver.rkt: Image ID obtained: %s", ID)
+
+			currApp.Image.ID = ID
+			currApp.Mounts = mounts
+			currApp.App.Environment = append(currApp.App.Environment, environment...)
+
+			// Add memory isolator
+			isolators := currApp.App.Isolators
+			memoryACIdentifier, err := appctypes.NewACIdentifier(appctypes.ResourceMemoryName)
+			if err != nil {
+				return nil, fmt.Errorf("[ERR] driver.rkt: failed to set ACIdentifier: %s", err)
+			}
+
+			if isolators.GetByName(*memoryACIdentifier) == nil {
+				s := fmt.Sprintf("%vM", task.Resources.MemoryMB)
+				resMemory, err := appctypes.NewResourceMemoryIsolator(s, s)
+				if err != nil {
+					return nil, fmt.Errorf("[ERR] driver.rkt: failed to create ResourceMemory: %s", err)
+				}
+
+				currApp.App.Isolators = append(isolators, resMemory.AsIsolator())
```

Comment:
You could just do `isolators = append(isolators, resMemory.AsIsolator())` here and for the CPU isolator, and after adding all the isolators do `currApp.App.Isolators = isolators`.